mii/deployment.py

```diff
@@ -2,94 +2,70 @@
-           enable_deepspeed=True,
-           enable_zero=False,
-           ds_config=None,
-           mii_config={},
-           version=1):
-    """Deploy a task using specified model. For usage examples see:
-
-        mii/examples/local/text-generation-example.py
-
-
-    Arguments:
-        task: Name of the machine learning task to be deployed.Currently MII supports the following list of tasks
-            ``['text-generation', 'text-classification', 'question-answering', 'fill-mask', 'token-classification', 'conversational', 'text-to-image']``
-
-        model: Name of a supported model for the task. Models in MII are sourced from multiple open-source projects
-            such as Huggingface Transformer, FairSeq, EluetherAI etc. For the list of supported models for each task, please
-            see here [TODO].
-
-        deployment_name: Name of the deployment. Used as an identifier for posting queries for ``LOCAL`` deployment.
-
-        deployment_type: One of the ``enum mii.DeploymentTypes: [LOCAL]``.
-            *``LOCAL`` uses a grpc server to create a local deployment, and query the model must be done by creating a query handle using
-              `mii.mii_query_handle` and posting queries using ``mii_request_handle.query`` API,
-
-        model_path: Optional: In LOCAL deployments this is the local path where model checkpoints are available. In AML deployments this
-            is an optional relative path with AZURE_MODEL_DIR for the deployment.
-
-        enable_deepspeed: Optional: Defaults to True. Use this flag to enable or disable DeepSpeed-Inference optimizations
-
-        enable_zero: Optional: Defaults to False. Use this flag to enable or disable DeepSpeed-ZeRO inference
-
-        ds_config: Optional: Defaults to None. Use this to specify the DeepSpeed configuration when enabling DeepSpeed-ZeRO inference
-
-        force_register_model: Optional: Defaults to False. For AML deployments, set it to True if you want to re-register your model
-            with the same ``aml_model_tags`` using checkpoints from ``model_path``.
-
-        mii_config: Optional: Dictionary specifying optimization and deployment configurations that should override defaults in ``mii.config.MIIConfig``.
-            mii_config is future looking to support extensions in optimization strategies supported by DeepSpeed Inference as we extend mii.
-            As of now, it can be used to set tensor-slicing degree using 'tensor_parallel' and port number for deployment using 'port_number'.
-
-        version: Optional: Version to be set for AML deployment, useful if you want to deploy the same model with different settings.
-    Returns:
-        If deployment_type is `LOCAL`, returns just the name of the deployment that can be used to create a query handle using `mii.mii_query_handle(deployment_name)`
-
-    """
-
-    # parse and validate mii config
+from .config import MIIConfig, DeploymentType
+
+
+def support_legacy_api(
```

Comment:
This should support all cases of the legacy API - however we may want to add a unit test. It depends if we plan to deprecate it soon or keep the old API around.

Reply:
In my opinion, we should stop supporting old APIs as soon as possible. We don't have many APIs and they are not closely related. It should be easy for users to migrate to new ones.